src/types/app-client.ts

```diff
@@ -339,6 +342,18 @@ export class ApplicationClient {
     const clearCompiled = await compileTeal(clear, this.algod)
     this._clearSourceMap = clearCompiled?.sourceMap
 
+    if (Config.debug && Config.projectRoot) {
+      persistSourceMaps({
```

Comment:
This means we cal compile twice, is that a problem?

Reply:
Is also is probably missing the deploy time control parameters? Can we instead pass through the source map from this method rather than calling compile?